plugins/inputs/tencentcloudcm/tencentcloudcm.go

```diff
@@ -0,0 +1,371 @@
+	"github.com/tencentcloud/tencentcloud-sdk-go/tencentcloud/common"
+	monitor "github.com/tencentcloud/tencentcloud-sdk-go/tencentcloud/monitor/v20180724"
+
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+// TencentCloudCM contains the configuration for the Tencent Cloud Cloud Monitor plugin.
+type TencentCloudCM struct {
+	Endpoint string `toml:"endpoint"`
+
+	Period config.Duration `toml:"period"`
+	Delay  config.Duration `toml:"delay"`
+
+	RateLimit int             `toml:"ratelimit"`
+	Timeout   config.Duration `toml:"timeout"`
+
+	DiscoveryInterval config.Duration `toml:"discovery_interval"`
+
+	Accounts []*Account `toml:"accounts"`
+
+	client cmClient
+
+	Log telegraf.Logger `toml:"-"`
+
+	windowStart  time.Time
+	windowEnd    time.Time
+	discoverTool *discoverTool
+}
+
+// Account defines a Tencent Cloud account
+type Account struct {
+	Name       string       `toml:"name"`
+	SecretID   string       `toml:"secret_id"`
+	SecretKey  string       `toml:"secret_key"`
+	Namespaces []*Namespace `toml:"namespaces"`
+
+	crs *common.Credential
+}
+
+// Namespace defines a Tencent Cloud CM namespace
+type Namespace struct {
+	Name    string    `toml:"namespace"`
+	Metrics []string  `toml:"metrics"`
+	Regions []*Region `toml:"regions"`
+}
+
+// Region defines a Tencent Cloud region
+type Region struct {
+	RegionName string              `toml:"region"`
+	Instances  []*monitor.Instance `toml:"instances"`
+}
+
+type metricObject struct {
+	Metric    string
+	Region    string
+	Namespace string
+	Account   *Account
+
+	isDiscovered bool
+
+	MonitorInstances []*monitor.Instance
+}
+
+type cmClient interface {
+	GetMetricObjects(t TencentCloudCM) []metricObject
+	NewClient(region string, crs *common.Credential, t TencentCloudCM) (monitor.Client, error)
+	NewGetMonitorDataRequest(namespace, metric string, instances []*monitor.Instance, t TencentCloudCM) *monitor.GetMonitorDataRequest
+	GatherMetrics(client monitor.Client, request *monitor.GetMonitorDataRequest, t TencentCloudCM) (*monitor.GetMonitorDataResponse, error)
+}
+
+// SampleConfig implements telegraf.Input interface
+func (t *TencentCloudCM) SampleConfig() string {
+	return `
+	## Endpoint to make request against, the correct endpoint is automatically
+	## determined and this option should only be set if you wish to override the
+	## default.
+	##   ex: endpoint = "tencentcloudapi.com"
+	# endpoint = ""
+  
+	# The default period for Tencent Cloud Cloud Monitor metrics is 1 minute (60s). However not all
+	# metrics are made available to the 1 minute period. Some are collected at
+	# 5 minute, 60 minute, or larger intervals.
+	# See: https://intl.cloud.tencent.com/document/product/248/33882
+	# Note that if a period is configured that is smaller than the default for a
+	# particular metric, that metric will not be returned by the Tencent Cloud API
+	# and will not be collected by Telegraf.
+	#
+	# Requested Tencent Cloud Cloud Monitor aggregation Period (required - must be a multiple of 60s)
```

Comment:
Comments should start with two hashes (`##`) according to telegraf rules. :-)

Reply:
fixed in commit 880ecd1